### ISSUE TITLE
Animate on function call of class

### ISSUE THREAD

**Issue by AgarwalSaurav** [jonobr1/two.js#394]
I wish to animate a particular object only when a particular function of a class is called.

In particular, I am trying to develop simple functions for visualization of algorithms. I have an array class which consists of rectangles. When a new element is added to the array I create a new rectangle and append to the list of rectangles in the array class. I wish to animate the newly added object.

See the function `addToArray` in https://github.com/AgarwalSaurav/algoanimator/blob/master/js/drawArray.js

Great library by the way. I am new to javascript. I am hoping the algorithm visualizer can be of great help.

Please feel free to ask if my question is not clear.

**Comment by jonobr1** [comment]
Thanks for posting. There are generally two ways to go about animating: distance based animations and keyframe based animations.

### 1. Distance Based Animations
From briefly looking at your code, it looks like you've implemented distance based animations already. Broadly speaking, distance based animations are great for quick and dirty state based animations. It requires one function to be called constantly, known as our animation frame in web development. With this function in place you can update a single value to change the state of an object. e.g:

````javascript
var two = new Two({
  fullscreen: true,
  autostart: true
}).appendTo(document.body);

var x = two.width / 2;
var y = two.height / 2;
var rectangle = two.makeRectangle(x, y, width, height);
var destination = new Two.Vector(x, y);

window.addEventListener('click', function(e) {
  destination.x = e.clientX;
  destination.y = e.clientY;
}, false);

var easing = 0.125; // A number between zero and one
two.bind('update', function(frameCount, timeDelta) {
  rectangle.translation.x += (destination.x - rectangle.translation.x) * easing;
  rectangle.translation.y += (destination.y - rectangle.translation.y) * easing;
});
````
In this simple example, the developer only needs to worry about what the value of the `destination` variable is in order to determine logic of the application. Two.js will automatically animate between the states. One down-side to distance based animations like this is that this may run faster or slower depending on the browser and hardware being used. You can offset this by multiplying the `easing` value to a normalized version of the `timeDelta` value offered in the `'update'` loop so that distance changes at a function of time, not just consistently for each frame.

### 2. Keyframe Based Animations
Alternatively, Keyframe Based Animations require you to specifically author state changes for key events and in Two.js require a third party library to be used in conjunction. For many of the demos and for my personal experience I learned with Tween.js ([link](https://github.com/tweenjs/tween.js/)), but there are many out there and by no means does Two.js force you to use this library alone. With a third party library, you create specific animations on page load or at runtime and then invoke those functions when needed. Here's the same example using Tween.js:

````javascript
var two = new Two({
  fullscreen: true,
  autostart: true
}).appendTo(document.body);

var x = two.width / 2;
var y = two.height / 2;
var rectangle = two.makeRectangle(x, y, width, height);
var keyframe = new TWEEN.Tween(rectangle.translation);

window.addEventListener('click', function(e) {
  var duration = 1000; // Defined in milliseconds
  var destination = {
    x: e.clientX,
    y: e.clientY
  };
  keyframe.to(destination, duration);
  keyframe.start();
}, false);

two.bind('update', function(frameCount, timeDelta) {
  TWEEN.update();
});
````

As you can see, these two models use very similar implementation techniques and can A) be interchangeable and B) be used in parallel. I often will have one attribute like position be keyframed based and then have another property like opacity be distance based.

Finally, these are two general purpose ways to approach animation, but there are many other ways including physics based animations. Hope this helps explain a bit of the landscape of animations for you. If you have any other specific questions regarding how to implement this in Two.js, feel free to open a new issue.

**Comment by AgarwalSaurav** [comment]
The keyframe based animations made it very convenient. I was able to implement comfortably. Thank you :)
